purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1899,6 +1899,58 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
             }
         }
 
+        /**
+         * Check if billing is supported for the current Play user (meaning IN-APP purchases are supported)
+         * and optionally, whether a specified feature type is supported. This method is asynchronous since
+         * it requires a connected BillingClient.
+         * @param context A context object that will be used to connect to the billing client
+         * @param feature An optional feature type to check for support. Must be one of [BillingFeature]
+         *                  If null, no specific feature support will be checked.
+         * @param callback Callback that will be notified when the check is complete.
+         */
+        @JvmStatic
+        fun canMakePayments(
+            context: Context,
+            feature: BillingFeature?,
+            callback: Callback<Boolean>
+        ) {
+            BillingClient.newBuilder(context)
+                .enablePendingPurchases()
+                .setListener { _, _ -> }
+                .build()
+                .let { billingClient ->
+                    billingClient.startConnection(
+                        object : BillingClientStateListener {
+                            override fun onBillingSetupFinished(billingResult: BillingResult) {
+                                try {
+                                    val featureSupportedResultOk = feature?.let {
```

Comment:
We don't really need to call isFeatureSupported if billingSupportedResultOk is false right? We could add a check for billingResult.isSuccessful before even checking if the feature is supported to avoid calling isFeatureSupported if not needed.